### PR TITLE
remove sources_for_hydro

### PR DESCRIPTION
**Pull request by zingale**
## PR summary

we don't need the sources_for_hydro MF.  We can add the predictor to the old sources
directly in src_to_prim

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
this is blocked by #1709

**Comment by zingale** [comment]
all tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-05-13-004/index.html

File: Source/driver/Castro_advance_ctu.cpp [hunk 1]
```diff
@@ -147,14 +134,10 @@ Castro::do_advance_ctu(Real time,
 #endif                
                       old_source, Sborder, S_new, prev_time, dt, apply_sources_to_state);
 
-      // Apply the old sources to the sources for the hydro.
-      // Note that we are doing an add here, not a copy,
-      // in case we have already started with some source
-      // terms (e.g. the source term predictor, or the SDC source).
+      if (do_hydro) {
```
Comment (maxpkatz) [inline Source/driver/Castro_advance_ctu.cpp:137]: Can you add a comment here to explain that this operation is just filling the ghost zones of Source_Type?

Comment (zingale) [reply Source/driver/Castro_advance_ctu.cpp:137]: done

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**